### PR TITLE
Throw error with name collision

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
In reference to #208, I have been thinking about it most of the day. To me it makes most sense to give an error instead of a warning. If we were to throw a warning you would have to recursively search for names that wouldn't collide, and that search space would be more and more obscure.

``` r
library(recipes)
library(dplyr)
data <- USArrests %>%
  mutate(PC1 = "collusion")  

rec <- recipe( ~ ., data = data)
pca_trans <- rec %>%
  step_center(all_numeric()) %>%
  step_scale(all_numeric()) %>%
  step_pca(all_numeric(), num = 3)

pca_estimates <- prep(pca_trans, training = data)
#> Error: Name collision occured in `step_pca()`. The following variable names already exists: PC1.
```

<sup>Created on 2018-10-02 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

### CODE REVIEW

File: R/misc.R [hunk 1]
```diff
@@ -457,3 +457,13 @@ simple_terms <- function(x, ...) {
   res
 }
 
+check_name <- function(x, y, name) {
```
Comment (topepo) [inline R/misc.R]: Maybe pass in the step object and use the first class to write the warning to avoid typos

**Comment by topepo** [comment]
Agreed on warning v  error. 

As an FYI we've been [talking about this in general](https://github.com/tidyverse/tibble/pull/469) in the tidyverse group so there will probably be a better way of handling this at the tibble-level in the near future.  

One change: can you make the specific column name changes via a function instead of repeated similar code? Maybe expand the checking code to include repair.

**Comment by topepo** [comment]
> Would you want it to repair instead of erroring?

My mistake. Error is fine (we'd wait for tibble changes for repair). I just want to removed repeated code.

**Comment by EmilHvitfeldt** [comment]
> My mistake.
No problem!

> Error is fine (we'd wait for tibble changes for repair).
Sure thing!

> I just want to removed repeated code.
Have I done that now? or have I overlooked something?

**Comment by topepo** [comment]
I was thinking that you have these repeated lines in a bunch of places:
```r
  newname <- names0(ncol(comps), object$prefix)
  check_name(colnames(newdata), newname, object)
  colnames(comps) <- newname
```
and they could be made into a function that can be called. If we ever need to change this code, we do it in one place.

**Comment by EmilHvitfeldt** [comment]
Okay I managed to remove some lines!

Unfortunately the naming scheme in `lag`, `depth`, `classdist` and `relu` doesn't use `prefix` and `names2` so I had to add a couple of arguments in check_name.

**Pull request closed without merging**